prow/pod-utils/clone/clone.go

```diff
@@ -239,14 +239,17 @@ func (g *gitCtx) commandsForBaseRef(refs prowapi.Refs, gitUserName, gitUserEmail
 		commands = append(commands, g.gitCommand("config", "http.cookiefile", cookiePath))
 	}
 
-	var depthArgs []string
+	var fetchOpts []string
+	if refs.BloblessFetch {
+		fetchOpts = append(fetchOpts, "--filter=blob:none")
+	}
 	if d := refs.CloneDepth; d > 0 {
-		depthArgs = append(depthArgs, "--depth", strconv.Itoa(d))
+		fetchOpts = append(fetchOpts, "--depth", strconv.Itoa(d))
 	}
 
 	if !refs.SkipFetchHead {
-		fetchArgs := []string{g.repositoryURI, "--tags", "--prune"}
-		fetchArgs = append(fetchArgs, depthArgs...)
+		fetchArgs := append([]string{}, fetchOpts...)
```

Comment:
Nit: I think using the `fetchOpts` naming is a bit confusing, because we already have `fetchArgs`. I think there are two options: (1) keep depthArgs as is and rename fetchOpts to filterArgs, or (2) combine fetchOpts and fetchArgs into one (keeping the fetchArgs name). I think option (2) is probably cleaner, but it may involve a larger refactoring which could be undesirable here for a purely additive change. So (1) seems better, but as this is a nit, up to you.

Reply:
My change originates in the way git documents fetch:
git fetch [<options>] [<repository> [<refspec>...]]

fetchOpts corresponds to <options> in the usage. It is defined once, separate from fetchArgs, since it is used again as you noted below. (Now technically "--tags" and "--prune" are options as well, but they don't apply later, so they are not included, thus the naming is not completely pure.)

Maybe it's best not to try to build a different list, and keep them categorized as depthArgs and filterArgs. I'll go that way, accepting it means extra append statements at each fetch invocation.

Reply:
Done